RecoEcal/EgammaClusterAlgos/src/CosmicClusterAlgo.cc

```diff
@@ -122,7 +121,7 @@ std::vector<reco::BasicCluster> CosmicClusterAlgo::makeClusters(
     
   }
   
-   sort(seeds.begin(), seeds.end(), EcalRecHitLess());
+   sort(seeds.begin(), seeds.end(), [](EcalRecHit x, EcalRecHit y){return x.energy() > y.energy();});
```

Comment:
in this and apparently many similar cases, I find this a bad coding pattern: a single named method is replaced by non-trivial inlined code [lambdas].
This is going to make the code much less maintainable.

Reply:
I think lambdas are appropriate if the function was only used once and defined in a completely different file as a functor. That spreads pieces that belong together a bit too far, for example in `[](auto& a, auto& b){return a.rawId() < b.rawId();}`. About the `EcalRecHitLess` I agree that it should better be defined in one place, right. How would you do it? Like 'isClusterEtLess' in a different file or maybe implemented as `operator<` in the RecHits? It is natural to compare them by energy after all.

Reply:
Thanks for the feedback!

Reply:
This comparison is needed/used for appropriate filling of the EcalRecHitCollection

Reply:
Ok very good! Then I'll reimplement the comparison function as before and put it also in that file, that seems to be a good place also for the sort comparators.

Reply:
Anyway, note  to self: `EcalRecHit` should be passed by reference...